src/dmd/link.d

```diff
@@ -120,35 +120,69 @@ version (Posix)
             static immutable(char*) nmeErrorMessage = "undefined reference to `_Dmain`";
         }
         FILE* stream = fdopen(fd, "r");
+        if (stream is null)
+            return -1;
+        return findStr(stream, nmeErrorMessage, stderr);
+    }
+
+    /*****************************
+     * Return `true` if the linker enables PIE by default.
+     */
+    private bool hasDefaultPIE(const char* cc)
+    {
+        OutBuffer cmd;
+        cmd.writestring(cc);
+        cmd.writestring(" -v 2>&1");
+        FILE* fp = popen(cmd.peekChars(), "r");
+        scope(exit) pclose(fp);
+        return 1 == findStr(fp, "--enable-default-pie");
+    }
```

Comment:
What's the overhead in opening this process on _every_ compilation? :thinking:

Reply:
The linker is slow anyway would be an argument. 
But maybe we can pessimize older cc version instead of the new ones i.e optimistically assume that `cc` can handle the `-no-pie` switch and retry if it fails.